doc/process-stages.md

```diff
@@ -0,0 +1,62 @@
+# Process Stages
+
+Eigenflow requires at least one stage to run the process.
+To define process stages create case objects which extend `ProcessStage` trait. For example:
+
+```scala
+case object StageName extends ProcessStage
+```
+
+The stages are mainly used as checkpoints during the process.
+`Eigenflow` persists each stage completion and the result of that stage.
+The result of stage execution must be returned in a `future`, to define stage execution logic use DSL:
+
+```scala
+StageName {
+  process()
+}
+```
+
+the `process` function in this example must return a `Future[A]`, then the next stage will receive A as argument.
+The `A` can be anything but serialization/deserialization implicit functions must be available:
+
+```scala
+A => String
+String => A
+```
+
+`Eigenflow` provides the `PrimitiveImplicits` object which can be imported to enable some primitives serialization.
+
+If you need access to process context, for example to calculate for which date to fetch the report you can use `withContext` method.
+Here is an example of downloading and archiving a report.
+
+```scala
+case object Initial extends ProcessStage
+case object Download extends ProcessStage
+case object Archive extends ProcessStage
+
+val init = Initial withContext { context: ProcessContext =>
+  Future {
+    context.processingDate.minusDays(1) // means always download report from yesterday (relating to the current processingDate)
```

Comment:
Hmm, according to #7 `processingDate` is not yet a `LocalDateTime`, so this particular line is not yet valid. Perhaps replace with `def dayBefore(date: Date) = ...; dayBefore(context.processingDate)`?

Also we'd need to add a comment on #7 reminding us to change this back to `context.processingDate minusDays 1`.